scylla/src/transport/topology.rs

```diff
@@ -799,6 +803,13 @@ async fn query_peers(conn: &Arc<Connection>, connect_port: u16) -> Result<Vec<Pe
     let peers_query_stream = conn
         .clone()
         .query_iter(peers_query)
+        .map(|it| {
+            it.and_then(|it| {
+                it.rows_stream::<NodeInfoRow>()
+                    .map_err(|err| RowsParseError::from(err).into())
+                    .map(|iter| iter.into_stream())
+            })
+        })
         .into_stream()
         .try_flatten()
         .and_then(|row_result| future::ok((NodeInfoSource::Peer, row_result)));
```

Comment:
Before we had just `.into_stream()`. now there is `into_stream()` x2 + `.rows_stream()`  - why?
There is also a lot of nesting and shadowing - it's quite difficult to read. Please make it more readable. Few ideas:
- Explicit types
- More descriptive variable names
- Intermediate variables and `if let`
- Definitely less nesting and indentation